### DISCUSSION TITLE
messages/media local cache

### DISCUSSION THREAD

**Discussion by cedricloneux** [ReagentX/imessage-exporter#99] [Q&A] [answered]
By default, are all iCloud messages downloaded locally?
I'm surprised that iCloud is indicating my messages use 22GB, but when I checked the 'chat.db' file size, it's only using 46MB, and attachments are 9.2GB.
This seems to confirm that the majority of messages and attachments are kept on iCloud and not cached locally.
I used these commands to check the file size

`du -sh ~/Library/Messages/chat.db`
`du -sh ~/Library/Messages/Attachments`

Does the command-line interface export all media, including those that may not be downloaded locally?

**Comment by ReagentX** [comment, marked as answer]
You asked this already: https://github.com/ReagentX/imessage-exporter/discussions/78

**Reply by cedricloneux** [reply]
Not really You am talking about attachements here and asking for solution

**Reply by ReagentX** [reply]
If the media are not downloaded locally, this software cannot export it.
